### PR TITLE
Fix: `lapack/dpttrf` compiles and runs correctly

### PR DESCRIPTION
**Pull request by Aakash788**
### Add integration test for DPTTRF subroutine

This PR adds an integration test for the DPTTRF subroutine as requested in issue #4499. The test verifies that the DPTTRF subroutine compiles and runs correctly.

- Adds `dpttrf.f` with the test code.
- Updates `CMakeLists.txt` to include the new test.

### CODE REVIEW

**Comment by kmr-srbh** [comment]
Thanks for this @Aakash788! The CI failure occurs due to #4504.

File: integration_tests/dpttrf.f [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+*     .. Scalar Arguments ..
+      INTEGER            INFO, N
+*     ..
+*     .. Array Arguments ..
+      DOUBLE PRECISION   D( * ), E( * )
+*     ..
+*
+*  =====================================================================
+*
+*     .. Parameters ..
```
Comment (Pranavchiku) [inline integration_tests/dpttrf.f:16]:
```suggestion
SUBROUTINE DPTTRF(N, D, E, INFO)
INTEGER            INFO, N
DOUBLE PRECISION   D( * ), E( * )
```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Let's rename test to `lapack_01.f90` which is free form version of Fortran

> lapack_01.f90

```fortran
subroutine dpttrf(n, d, e, info)
integer :: n, info
double precision :: d( * ), e( * )
double precision :: zero
parameter( zero = 0.0d+0 )
integer :: i, i4
double precision :: ei
intrinsic :: mod
info = 0
if( n.eq.0 ) return
i4 = mod( n-1, 4 )
do 10 i = 1, i4
   if( d( i ).le.zero ) then
      info = i
      go to 30
   end if
   ei = e( i )
   e( i ) = ei / d( i )
   d( i+1 ) = d( i+1 ) - e( i )*ei
10 continue
do 20 i = i4 + 1, n - 4, 4
   if( d( i ).le.zero ) then
      info = i
      go to 30
   end if
   ei = e( i )
   e( i ) = ei / d( i )
   d( i+1 ) = d( i+1 ) - e( i )*ei
   if( d( i+1 ).le.zero ) then
      info = i + 1
      go to 30
   end if
   ei = e( i+1 )
   e( i+1 ) = ei / d( i+1 )
   d( i+2 ) = d( i+2 ) - e( i+1 )*ei
   if( d( i+2 ).le.zero ) then
      info = i + 2
      go to 30
   end if
   ei = e( i+2 )
   e( i+2 ) = ei / d( i+2 )
   d( i+3 ) = d( i+3 ) - e( i+2 )*ei

   if( d( i+3 ).le.zero ) then
      info = i + 3
      go to 30
   end if
   ei = e( i+3 )
   e( i+3 ) = ei / d( i+3 )
   d( i+4 ) = d( i+4 ) - e( i+3 )*ei
20 continue
if( d( n ).le.zero ) info = n

30 continue
return

end subroutine

program lapack_01
   integer :: n = 3, info
   double precision :: d(3), e(2)
   d = [4, 5, 6]
   e = [1, 2]
   call dpttrf(n, d, e, info)
   print *, "d: ", d
   print *, "e: ", e
   print *, "info: ", info
end program
```

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -1494,6 +1494,7 @@ RUN(NAME dealloc_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc EXTRA_ARGS --r
 RUN(NAME dealloc_02 LABELS gfortran)
 
 RUN(NAME double_complex_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
+RUN(NAME dpttrf LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
```
Comment (Pranavchiku) [inline integration_tests/CMakeLists.txt:1497]:
```suggestion
RUN(NAME lapack_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
```

**Comment by Pranavchiku** [comment]
With these changes, I think this PR will be good to merge.

File: integration_tests/lapack_01.f90 [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+program lapack_01
+   integer :: n = 3, info
+   double precision :: d(3), e(2)
+   d = [4, 5, 6]
+   e = [1, 2]
+   call dpttrf(n, d, e, info)
+   print *, "d: ", d
+   print *, "e: ", e
+   print *, "info: ", info
```
Comment (Pranavchiku) [inline integration_tests/lapack_01.f90:67]:
```suggestion
   print *, "sum(d): ", sum(d)
   if ( abs(sum(d) - 13.907894736842106) > 1e-6 ) error stop
   print *, "sum(e): ", sum(e)
   if ( abs(sum(e) - 0.67105263157894735) > 1e-7 ) error stop
   print *, "info: ", info
   if ( info /= 0 ) error stop
```

One last change.

**Review by Pranavchiku — APPROVED** [review]

**Review by certik — APPROVED** [review]
Thanks!

**Pull request merged**